python-runner/server.py

```diff
@@ -0,0 +1,327 @@
+#!/usr/bin/env python3
+"""
+Python execution server for Metabase.
+Runs Python scripts in isolated processes with resource limits and monitoring.
+"""
+
+import os
+import sys
+import time
+import signal
+import resource
+import threading
+import subprocess
+from pathlib import Path
+from queue import Queue, Empty
+from datetime import datetime, timedelta
+from typing import Optional, Dict, Any
+
+from flask import Flask, request, jsonify
+from werkzeug.exceptions import HTTPException
+
+DEFAULT_TIMEOUT = 30  # seconds
+MAX_MEMORY_MB = 1024  # Increased for pandas/numpy
+MAX_CPU_TIME_SECONDS = 60
+MAX_QUEUE_SIZE = 10
+QUEUE_TIMEOUT = 60  # seconds to wait in queue
+MAX_FILE_DESCRIPTORS = 256
+MAX_PROCESSES = 50
+
+app = Flask(__name__)
+
+metrics = {
+    "total_requests": 0,
+    "successful_executions": 0,
+    "failed_executions": 0,
+    "timeouts": 0,
+    "rejected_requests": 0,
+    "current_queue_size": 0,
+    "average_execution_time": 0,
+    "execution_times": [],
+}
+metrics_lock = threading.Lock()
```

Comment:
There's probably an atomic dict or something we can use.

This python is still kinda slop even with a bunch of steering, could use some extra passes or a rewrite next week.